azure-pipelines.yml

```diff
@@ -11,12 +11,18 @@ variables:
   value: Xamarin.Forms.sln
 - name: BuildVersion
   value: $[counter('$(Build.SourceBranchName)_counter', 1)]
-- name: MONO_VERSION
-  value: 5_18_1
 - name: XCODE_VERSION
   value: 10.2
 - name: NUGET_VERSION
-  value: 5.0.2
+  value: 5.2.0
+- name: DOTNET_VERSION
```

Comment:
I wonder if we should pin the variables here on in AzureDev Ops

> You can choose which variables are allowed to be set at queue time and which are fixed by the pipeline author. If a variable appears in the variables block of a YAML file, it is fixed and cannot be overridden at queue time. To allow a variable to be set at queue time, make sure it doesn't appear in the variables block of a pipeline or job. You can set a default value in the editor, and that value can be overridden by the person queuing the pipeline.

Reply:
So that's why whenever I set these they never change.

Yea I think not pinning these here would be good or fixing the code here so that this value is overridable

Like rename this to *DOTNET_VERSION_DEFAULT*

And then if there's an environment variable set for *DOTNET_VERSION* then we use that otherwise we use the default.

Is there a way to set a variable conditionally if it's not specified by the pipeline? Or is that just something the scripts and YAML tasks will need to account for?

Reply:
alright @rmarinho until this gets resolved

https://developercommunity.visualstudio.com/content/problem/741877/variables-and-parameters-are-not-being-expanded-in.html

Let's just leave this as is

in theory it's setup to work so once the issue above is resolved then we can specify versions as parameters